crates/txtx-core/src/validation/hcl_validator/block_processors.rs

```diff
@@ -0,0 +1,315 @@
+//! Block processing for HCL validation.
+
+use std::collections::HashMap;
+
+use txtx_addon_kit::hcl::{structure::{Block, BlockLabel}, Span};
+use txtx_addon_kit::constants::{
+    DESCRIPTION, DEPENDS_ON, MARKDOWN, MARKDOWN_FILEPATH, POST_CONDITION, PRE_CONDITION,
+};
+
+use crate::kit::types::commands::CommandSpecification;
+use crate::runbook::location::SourceMapper;
+use crate::validation::hcl_validator::visitor::{
+    CollectedItem, DefinitionItem, DeclarationItem, BlockType, Position,
+    ValidationError,
+};
+
+/// Extract position from a block's identifier span
+///
+/// Converts the block's identifier span to a Position using the source mapper.
+/// Returns a default position (1, 1) if the block has no span information.
+///
+/// This helper consolidates the repeated pattern of extracting positions
+/// from block identifiers, reducing code duplication across processor functions.
+fn extract_block_position(block: &Block, source_mapper: &SourceMapper) -> Position {
+    block.ident.span()
+        .as_ref()
+        .map(|span| {
+            let (line, col) = source_mapper.span_to_position(span);
+            Position::new(line, col)
+        })
+        .unwrap_or_default()
+}
+
+/// Process a block during the collection phase.
+pub fn process_block(
+    block: &Block,
+    block_type: BlockType,
+    addon_specs: &HashMap<String, Vec<(String, CommandSpecification)>>,
+    source_mapper: &SourceMapper,
+) -> Result<Vec<CollectedItem>, ValidationError> {
+    match block_type {
+        BlockType::Signer => process_signer(block),
+        BlockType::Variable => process_variable(block, source_mapper),
+        BlockType::Output => process_output(block),
+        BlockType::Secret => process_secret(block),
+        BlockType::Action => process_action(block, addon_specs, source_mapper),
+        BlockType::Flow => process_flow(block, source_mapper),
+        BlockType::Addon | BlockType::Unknown => Ok(Vec::new()),
```

Comment:
Currently, the `Action` processing allows actions from any `Addon`, but maybe a future version of this could first index the `Addon`s and only allow an action if the user has declared that addon.